src/Components/test/E2ETest/ServerExecutionTests/CircuitGracefulTerminationTests.cs

```diff
@@ -34,63 +32,65 @@ public class CircuitGracefulTerminationTests : BasicTestAppTestBase, IDisposable
 
+        public override async Task InitializeAsync()
+        {
+            // These tests manipulate the browser in  ways that make it impossible to use the same browser
+            // instance across tests (One of the tests closes the browser). For that reason we simply create
+            // a new browser instance for every test in this class sos that there are no issues when running
+            // them together.
+            await base.InitializeAsync(Guid.NewGuid().ToString());
+        }
+
         protected override void InitializeAsyncCore()
         {
-            // On WebAssembly, page reloads are expensive so skip if possible
-            Navigate(ServerPathBase, _serverFixture.ExecutionMode == ExecutionMode.Client);
+            Navigate(ServerPathBase, noReload: false);
             MountTestComponent<CounterComponent>();
             Browser.Equal("Current count: 0", () => Browser.FindElement(By.TagName("p")).Text);
-        }
 
-        [Fact]
-        public async Task ReloadingThePage_GracefullyDisconnects_TheCurrentCircuit()
-        {
-            // Arrange
             GracefulDisconnectCompletionSource = new TaskCompletionSource<object>(TaskContinuationOptions.RunContinuationsAsynchronously);
             Sink = _serverFixture.Host.Services.GetRequiredService<TestSink>();
             Messages = new List<(Extensions.Logging.LogLevel level, string eventIdName)>();
-            Sink.MessageLogged += wc => Log(wc);
+            Sink.MessageLogged += Log;
+        }
 
-            // Act
+        [Fact]
+        public async Task ReloadingThePage_GracefullyDisconnects_TheCurrentCircuit()
+        {
+            // Arrange & Act
             _ = ((IJavaScriptExecutor)Browser).ExecuteScript("location.reload()");
             await Task.WhenAny(Task.Delay(10000), GracefulDisconnectCompletionSource.Task);
 
             // Assert
             Assert.Contains((Extensions.Logging.LogLevel.Debug, "CircuitTerminatedGracefully"), Messages);
+            Assert.Contains((Extensions.Logging.LogLevel.Debug, "CircuitDisconnectedPermanently"), Messages);
         }
 
         [Fact]
         public async Task ClosingTheBrowserWindow_GracefullyDisconnects_TheCurrentCircuit()
         {
-            // Arrange
-            GracefulDisconnectCompletionSource = new TaskCompletionSource<object>(TaskContinuationOptions.RunContinuationsAsynchronously);
-            Sink = _serverFixture.Host.Services.GetRequiredService<TestSink>();
-            Messages = new List<(Extensions.Logging.LogLevel level, string eventIdName)>();
-            Sink.MessageLogged += wc => Log(wc);
-
-            // Act
+            // Arrange & Act
             Browser.Close();
+            // Set to null so that other tests in this class can create a new browser if necessary so
+            // that tests don't fail when running together.
+            Browser = null;
```

Comment:
Is this also necessary? Doesn't the isolation context take care of this?

Reply:
Yeah, it's not required. Its lefover from a previous try.